src/github_runner_manager/openstack_cloud/openstack_runner_manager.py

```diff
@@ -582,18 +582,17 @@ def _check_state_and_flush(self, instance: OpenstackInstance, busy: bool) -> Non
             # kill both Runner.Listener and Runner.Worker processes.
             # This kills pre-job.sh, a child process of Runner.Worker.
             kill_command = (
-                f"pgrep -x {RUNNER_LISTENER_PROCESS} "
-                f"&& kill $(pgrep -x {RUNNER_LISTENER_PROCESS});"
-                f"pgrep -x {RUNNER_WORKER_PROCESS} "
-                f"&& kill $(pgrep -x {RUNNER_WORKER_PROCESS});"
+                f"pgrep -x {RUNNER_LISTENER_PROCESS} && "
+                f"kill $(pgrep -x {RUNNER_LISTENER_PROCESS});"
+                f"pgrep -x {RUNNER_WORKER_PROCESS} && kill $(pgrep -x {RUNNER_WORKER_PROCESS});"
             )
         else:
             logger.info(
                 "Attempting to kill runner process on %s if not busy", instance.server_name
             )
             # Only kill Runner.Listener if Runner.Worker does not exist.
             kill_command = (
-                f"pgrep -x {RUNNER_WORKER_PROCESS} || pgrep -x {RUNNER_LISTENER_PROCESS} && "
+                f"! pgrep -x {RUNNER_WORKER_PROCESS} && pgrep -x {RUNNER_LISTENER_PROCESS} && "
                 f"kill $(pgrep -x {RUNNER_LISTENER_PROCESS})"
             )
```

Comment:
Since we had issues with this, could we at least log a bit more here? I think go ahead with this change, just add a follow up ticket to:
* Log the output of the individual commands
* * Potentially move more of the logic to Python rather than in the bash command (e.g., instead of pgrep, just get all the processes and do the logic checks in Python)

Reply:
The reason for the pgrep was to elimiate as much race-conditions due to issuing commands over the network (from `ps aux` check to the time we actually determine something may already have discrepancies on the state of the runner).
I've added logging of the outputs though, thank you for the review!